types/random-gradient/tsconfig.json

```
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6"],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": ["../"],
+        "types": [],
+        "noEmit": true,
+        "esModuleInterop": true,
```

```suggestion
        "noEmit": true,
```
It's required for client consumption here, as this is regular CommonJS module:
https://github.com/bukinoshita/random-gradient/blob/master/index.js#L7
Here (on DT), we tests for all scenarios, so DT package should support older syntax as well:
```ts
import gradient ='require('random-gradient');
```
It won't impact client consuming this DT definition later.

---

I think, "esModuleInterop" property is required because so this module can be used with new import syntax

---

Yes, it can, but when using it, I fully agree. Here it's advised t not use it and make your package
work accross the TypeScript different versions spectrum:
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

---

OK, I removed it